### ISSUE TITLE
Pencil jumps above grabbed position

### ISSUE THREAD

**Issue by KatieWoe** [phetsims/mean-share-and-balance#315]
**Test device**
Samsung
**Operating System**
Win 11
**Browser**
Firefox
**Problem description**
For https://github.com/phetsims/qa/issues/1105
On the Level Out Screen, the first time you grab the Predict Mean pencil, it jumps up above where you are grabbing, making control difficult. It behaves normally after, until the reset all button is used. It doesn't seem to happen on other screens.

**Visuals**
![flyingpencil](https://github.com/phetsims/mean-share-and-balance/assets/41024075/a944bfad-3ee0-4cec-9dcf-e7d62ce3d607)

Troubleshooting information:

<details>

!!!!! DO NOT EDIT !!!!!
Name: ‪Mean: Share and Balance‬
URL: https://phet-dev.colorado.edu/html/mean-share-and-balance/1.1.0-dev.7/phet/mean-share-and-balance_all_phet.html
Version: 1.1.0-dev.7 2024-07-08 22:39:22 UTC
Features missing: applicationcache, applicationcache, touch
Flags: pixelRatioScaling
User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:128.0) Gecko/20100101 Firefox/128.0
Language: en-US
Window: 1536x731
Pixel Ratio: 1.25/1
WebGL: WebGL 1.0
GLSL: WebGL GLSL ES 1.0
Vendor: Mozilla (ANGLE (Intel, Intel(R) HD Graphics Direct3D11 vs_5_0 ps_5_0), or similar)
Vertex: attribs: 16 varying: 30 uniform: 4096
Texture: size: 16384 imageUnits: 16 (vertex: 16, combined: 32)
Max viewport: 32767x32767
OES_texture_float: true
Dependencies JSON: {}

</details>

**Comment by KatieWoe** [comment]
Update: It does happen on other screens. It does not happen right away.

Steps: 
1. Turn on Predict Mean and move the pencil
2. Reset all
3. Turn on Predict Mean and try to move the pencil again.

**Comment by marlitas** [comment]
I fixed this with a bidirectional Dynamic Property, but I got very confused in the process as well.

The problem was pretty clear: 
During reset we were resetting the `valueProperty` in the model, however the `Vector2Property` that was created for the drag listener in the accessible slider was not being reset. Therefore, when we reset and go to click on the slider again, it's `valueProperty` is overwritten by the drag listener's `Vector2Property`. 

What I'm confused about is that this pattern is used in CAV `PredictionSlider` and I cannot find any bugs... I have compared the code in all different angles and I don't understand why the CAV and MSAB implementations are behaving differently.

This is ready for review, and I would love @jbphet to take a look at the CAV code so that I can better understand why it is working in CAV but was buggy for MSAB along with reviewing the fix committed above.

**Comment by jbphet** [comment]
I started reviewing the code in `MeanPredictionHandle`, and it seemed to me that there was a simpler way to do what needed to be done.  Specifically, it seemed like the intermediate property could be removed and the drag listener could just set the value of the managed property directly.  This solves the reset problem that motivated the issue, simplifies the code, and reduces the file size by about 10%.  That all makes me suspicious I'm being naive and have missed something.  So, I have a couple questions for @marlitas:

- Does the approach that I've committed seem reasonable to you, or am I missing something?
- You mentioned that this is the pattern used in CAV.  I didn't look at that yet, since I thought I'd try this out.  If the simplification looks viable, should we revisit CAV?
